VetClinic.DAL/Repositories/Base/Repository.cs

```diff
@@ -0,0 +1,144 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using VetClinic.Core.Interfaces.Repositories.Base;
+using VetClinic.DAL.Context;
+
+namespace VetClinic.DAL.Repositories.Base
+{
+    public class Repository<TEntity> : IRepository<TEntity>
+        where TEntity : class
+    {
+        private readonly FakeDbContext _context;
+        public Repository(FakeDbContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Gets a list of elements that satisfy the specific condition
+        /// </summary>
+        /// <param name="filter">A condition for selecting elements</param>
+        /// <param name="orderBy">A function to order elements</param>
+        /// <param name="include">A function to include navigation properties</param>
+        /// <param name="asNoTracking"><c>false</c>To disable changing tracking; Otherwise, <c>true</c></param>
+        /// <returns>The list of elements that satisfy the condition specified by <paramref name="filter"/>.</returns>
+        public async Task<IList<TEntity>> GetAsync(
+            Expression<Func<TEntity, bool>> filter = null,
+            Func<IQueryable<TEntity>,
+                IOrderedQueryable<TEntity>> orderBy = null,
+            Func<IQueryable<TEntity>,
+                IIncludableQueryable<TEntity, object>> include = null,
+            bool asNoTracking = false)
+        {
+            IQueryable<TEntity> query = GetConfiguredSelection(filter, include, asNoTracking);
+            //consider using ternary operator
+            if(orderBy != null)
+            {
+                return await orderBy(query).ToListAsync();
+            }
+            return await query.ToListAsync();
+        }
+
+        public async Task<TEntity> GetByIdAsync(object id)
+        {
+            return await _context.Set<TEntity>().FindAsync(id);
```

Comment:
What happened if no find?

Reply:
> What happened if no find?

been implementing this method thinking that even tho it doesn't return any element it is a responsibility of controller to check if there is any element found (then return the according status code: not found, ok, etc)